### PR TITLE
Add agent_id_available probe for pre-registration availability check

### LINKED ISSUE #41: agent_id 가용성 사전 체크 API 필요
## 문제

`registerClient` mutation은 `allowedAgentIds`를 확정해서 CLIENT_TOKEN을 발급하지만, 해당 `agent_id`가 이미 다른 지갑에 의해 `agent_policies`에 등록돼있는지 **사전에 확인할 수 있는 방법이 없다**.

결과적으로 충돌은 `vicoop-client`의 첫 WS register 시점에야 감지됨 (`ws.ts:57` — `'agent id owned by a different wallet'`). 이 시점이 되면 이미 CLIENT_TOKEN은 발급돼있고, `allowedAgentIds` 변경은 어렵거나 불가능해서 **토큰 재발급이 필요한 비싼 실패 모드**.

## 현재 가용한 수단의 한계

- `agentPolicyByAgentId(agentId: "...")` via GraphQL — RLS(`agent_policies_select`)로 `owner_wallet = current_wallet OR is_admin()`만 SELECT 가능. 다른 지갑이 소유한 id를 조회하면 `null` 반환 → **'비어있음'과 '남이 소유함'을 구분할 수 없음**.
- `allAgentPolicies` — 마찬가지로 RLS로 내 소유만 나옴.
- admin agent의 `list_active_agents` 툴 — 현재 WS로 연결된 agent만, non-admin은 자기 것만.
- 수비적 네이밍 (`<project>-openclaw-<uuid>`) — 실용적 회피책이지만 근본 해결은 아님.

## 제안

소유자/세부 정보를 노출하지 않고 **boolean availability**만 돌려주는 공개 조회 엔드포인트 하나. 예:

```graphql
query { agentIdAvailable(agentId: \"openclaw-local\") }  # → true/false
```

또는 REST:
```
GET /agents/{id}/availability  # 200 {\"available\": bool}
```

### 보안 고려

- **소유자/owner_wallet 등 메타데이터는 절대 노출하지 않음** — boolean만.
- Enumeration(사전식 정찰) 완화를 위해 IP당 레이트 리밋 추천.
- 인증 요구할지는 선택 — 공개가 더 편하지만, `vbc_caller_*` 토큰 요구하면 남용 억제 효과.
- `is_admin()` 통해서는 owner_wallet 포함 전체 조회 계속 가능 (현재 동작 유지).

### 동작 정의

`taken`은 다음 중 하나면 true:
- `agent_policies`에 해당 `agent_id` 행이 존재 (WS 접속 후 영속)
- `registry.agents` 메모리 맵에 현재 접속 중 (WS 이미 점유 중, 아직 policy 행은 있을 수 있음)

실제 구현은 persistent(`agent_policies`) 체크면 충분 — in-memory는 DB 행을 전제로 만들어짐.

## 참고

- `packages/server/src/ws.ts:38-60` `ensureAgentPolicy` — 충돌이 관찰되는 지점
- `packages/server/schema.sql:430-433` `agent_policies_select` RLS 정책
- UX 맥락: #17, #21 (client 원라이너 설치 흐름) — 첫 설치 경험에서 agent_id 네이밍이 자연스러운 선택이 되도록

### PR DESCRIPTION
**Pull request by longfin**
## Summary

- Add `agent_id_available(text) returns boolean` SQL function to `packages/server/schema.sql`, exposed via PostGraphile as `Query.agentIdAvailable(agentId)`.
- `SECURITY DEFINER` bypasses the `agent_policies_select` RLS so the probe is authoritative across all owners, but returns **only a boolean** — never `owner_wallet` or other metadata.
- Hardened definer context: `search_path = pg_catalog, public` (no `pg_temp`, schema-qualified `public.agent_policies`) and function ownership explicitly transferred to `app_postgraphile` so the migration-time superuser context doesn't leak into runtime. `app_postgraphile` already carries the RLS-bypass policy on `agent_policies`, which is the minimum required.
- Explicit NULL/empty input guard — raises `invalid_parameter_value` so a plainly invalid id can't be reported as "available".
- Restricted to `app_authenticated` (`vbc_caller_*` token required) to mitigate enumeration. Admin `is_admin()` paths retain full-row visibility unchanged.
- Add DB-gated tests: free id, taken id, cross-wallet authenticated caller (proves RLS bypass works — otherwise a different wallet would see the row as absent and wrongly report `true`), `app_anonymous` permission denied (locks in the enumeration mitigation), NULL/empty input rejection.

## Why

`registerClient()` accepts `allowed_agent_ids` but does not verify the ids are free. A collision with another wallet only surfaces at first WS register (`packages/server/src/ws.ts` `ensureAgentPolicy` → `'agent id owned by a different wallet'`), by which point the `CLIENT_TOKEN` is already issued and rotation/re-issue is needed. Meanwhile `agent_policies_select` RLS hides foreign-owned rows, so a non-admin direct GraphQL query can't tell `'free'` from `'taken by someone else'`. This probe closes that gap cheaply without leaking ownership.

Fixes #41.

## Follow-ups (out of scope)

- IP/caller rate limiting on GraphQL — mentioned in the issue but is a cross-cutting concern beyond this function.
- Wiring the probe into the admin-ui registration flow.

## Test plan

- [x] `pnpm --filter @vicoop-bridge/server typecheck`
- [x] `DATABASE_URL=... pnpm --filter @vicoop-bridge/server exec tsx --test 'src/**/*.test.ts'` — all green locally, including 5 new cases on `agent_id_available`
- [ ] Reviewer sanity-checks the SECURITY DEFINER posture: `search_path = pg_catalog, public` (no `pg_temp`), schema-qualified `public.agent_policies`, owner `app_postgraphile`
- [ ] Reviewer confirms `app_authenticated`-only GRANT matches desired enumeration posture (vs. opening to `app_anonymous`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/server/src/agent-id-available.test.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      // Once cleaned up, the id is free again.
+      assert.equal(await callAvailable(sql, agentId), true);
+    } finally {
+      await sql.end();
+    }
+  },
+);
+
+test(
+  'authenticated caller from a different wallet still sees taken=true (RLS bypass)',
```
[the remark this answers is not in the document]
Comment (longfin) [reply packages/server/src/agent-id-available.test.ts]: Fixed in 15ddf9e — renamed to "authenticated caller from a different wallet still sees available=false (RLS bypass)" to match the assertion.

File: packages/server/schema.sql [hunk 2]
```diff
@@ -457,6 +457,38 @@ CREATE POLICY agent_policies_postgraphile ON agent_policies
...
+-- id is claimable.
+CREATE OR REPLACE FUNCTION agent_id_available(agent_id TEXT)
+  RETURNS BOOLEAN
+  LANGUAGE plpgsql STABLE
+  SECURITY DEFINER
+  SET search_path = public, pg_temp
+AS $$
+BEGIN
+  RETURN NOT EXISTS(
+    SELECT 1 FROM agent_policies ap
```
[the remark this answers is not in the document]
Comment (longfin) [reply packages/server/schema.sql]: Fixed in b0dd315 — dropped pg_temp from search_path (now `pg_catalog, public`) and schema-qualified the `public.agent_policies` reference so a temp-object shadow cannot hijack resolution.

File: packages/server/schema.sql [hunk 3]
```diff
@@ -457,6 +457,42 @@ CREATE POLICY agent_policies_postgraphile ON agent_policies
...
+  -- with pg_temp in the path any role that can CREATE TEMP could shadow an
+  -- unqualified reference and hijack a definer-privileged resolution. The
+  -- function schema-qualifies agent_policies for the same reason.
+  SET search_path = pg_catalog, public
+AS $$
+BEGIN
+  RETURN NOT EXISTS(
+    SELECT 1 FROM public.agent_policies ap
+    WHERE ap.agent_id = agent_id_available.agent_id
+  );
```
[the remark this answers is not in the document]
Comment (longfin) [reply packages/server/schema.sql:497]: Fixed in 9ac9336 — added `ALTER FUNCTION agent_id_available(TEXT) OWNER TO app_postgraphile`. That role already has the RLS-bypass policy on `agent_policies`, which is the minimum needed here; migration-time superuser context no longer leaks into runtime.

[the remark this answers is not in the document]
Comment (longfin) [reply packages/server/schema.sql:497]: Fixed in 9ac9336 — added explicit NULL/empty guard that raises `invalid_parameter_value`. Opted for a hard error over silent false/null so bad inputs fail loudly at the call site.

File: packages/server/src/agent-id-available.test.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        `;
+
+        const available = await sql.begin(async (tx) => {
+          await tx`SET LOCAL ROLE app_authenticated`;
+          await tx`SELECT set_config('jwt.claims.wallet_address', ${other}, true)`;
+          const rows = await tx<{ available: boolean }[]>`
+            SELECT agent_id_available(${agentId}) AS available
+          `;
+          return rows[0]!.available;
+        });
```
[the remark this answers is not in the document]
Comment (longfin) [reply packages/server/src/agent-id-available.test.ts:96]: Fixed in 9ac9336 — added a DB-gated test that `SET LOCAL ROLE app_anonymous` and asserts `permission denied`. Also added a test for the new NULL/empty guard. Both lock in the surface so future grant or body changes fail loudly.

File: packages/server/schema.sql [hunk 5]
```diff
@@ -457,6 +457,57 @@ CREATE POLICY agent_policies_postgraphile ON agent_policies
...
+CREATE OR REPLACE FUNCTION agent_id_available(agent_id TEXT)
+  RETURNS BOOLEAN
+  LANGUAGE plpgsql STABLE
+  SECURITY DEFINER
+  -- pg_catalog first, public second, and pg_temp is intentionally absent:
+  -- with pg_temp in the path any role that can CREATE TEMP could shadow an
+  -- unqualified reference and hijack a definer-privileged resolution. The
+  -- function schema-qualifies agent_policies for the same reason.
+  SET search_path = pg_catalog, public
+AS $$
```
[the remark this answers is not in the document]
Comment (longfin) [reply packages/server/schema.sql:484]: Updated the PR description to reflect the final search_path (`pg_catalog, public`, no `pg_temp`) and the schema-qualified `public.agent_policies` reference. Also documented the `app_postgraphile` ownership transfer and NULL/empty guard added in subsequent commits.
